waspc/src/Wasp/Generator/Start.hs

```diff
@@ -5,19 +5,27 @@ where
 
 import Control.Concurrent (newChan)
 import Control.Concurrent.Async (concurrently, race)
+import Control.Monad.Except (ExceptT, MonadError (throwError), MonadIO (liftIO), runExceptT)
 import StrongPath (Abs, Dir, Path')
 import Wasp.Generator.Common (ProjectRootDir)
+import Wasp.Generator.DbGenerator (checkIfDbSchemaMatchesChecksum)
 import Wasp.Generator.Job.IO (readJobMessagesAndPrintThemPrefixed)
 import Wasp.Generator.ServerGenerator.Start (startServer)
 import Wasp.Generator.WebAppGenerator.Start (startWebApp)
 
--- | This is a blocking action, that will start the processes that run web app and server.
+-- | This calls a blocking action that will start the processes that run web app and server.
 --   It will run as long as one of those processes does not fail.
 start :: Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
 start projectDir = do
-  chan <- newChan
+  runExceptT $ do
+    checkIfDbSchemaMatchesChecksum projectDir
+    start' projectDir
+
+start' :: Path' Abs (Dir ProjectRootDir) -> ExceptT String IO ()
```

Comment:
The main reason I changed this type to `ExceptT String IO ()` and not `compileIO` used in Watch.hs was this was now a helper local to this file and it made it easier to get the short circuiting. `compileIO` was used in lots of areas and it was better to keep that `IO (Either String ())` in my book. But again, I'm unsure where the line is on these things so welcome feedback. Thanks!

Reply:
And to recap the problem without resorting to `ExceptT`, I have two or more `IO (Either String ())` functions and would like short circuiting behavior.

Reply:
No longer updating Start.hs, as the check now happens in Generator.